### PR TITLE
New design

### LINKED ISSUE #138: Cooler design
Dominik mention something like material.io could make the site look even cooler. Let's give it a go.

### PR DESCRIPTION
**Pull request by kdmnk**
**Framework: https://materializecss.com/**, that uses Material Design created by Google

- materialize.css generated with a sass generator, with custom colors for EC and other modifications (colors can be easily changed in components/color-variables.scss and variables.scss)
- modified materialize.js (for searchable select): https://codepen.io/yassinevic/pen/eXjqjb?editors=1111
- new logo (may be changed, the other pictures are still in the directory, and we can even add more, because i don't like that this logo does not have the 'ec' colors :D )
- mobile friendly (Tabulators may not fit in properly, but they are scrollable)
- separate admin blades
- changed folder structure
- added user page
- included some translations also
- included  #172 / #174 (more info on print page)
- included #156 / #164 (report a bug + with notification in case that browser does not support this function)
-----------
- please test every button and input-field (with incorrect values also) if everything works as expected. (also make sure if input's 'id' and label's 'for' tag are paired correctly) 
- don't forget to clear your browser cache and also warn the users about this after releasing...
-----------
Fixes #138

### CODE REVIEW

File: app/Http/Controllers/Auth/RegisterController.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ protected function validator(array $data)
             'place_of_birth' => ['required', 'string', 'max:255'],
             'date_of_birth' => ['required', 'date_format:Y-m-d'],
             'mothers_name' => ['required', 'string', 'max:255'],
-            'phone_number' => ['required', 'string', 'min:16', 'max:18'],
+            'phone_number' => ['required', 'string', 'min:11', 'max:18'],
```
Comment (katkodominik) [inline app/Http/Controllers/Auth/RegisterController.php]: Currently it does not format phone numbers automatically...

File: public/js/site.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//TODO rewrite for materialize modal
```
Comment (katkodominik) [inline public/js/site.js]: Currently I deleted the confirmation popups for the buttons in tabulators, because the bootstrap modals and materialize modals glitched together. If necessary, need to rewrite in a future issue, but this is mostly used by admins.

File: resources/views/admin/internet/mac_addresses/list.blade.php [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div id="internet-accesses-table"></div>
+<div id="mac-addresses-table"></div>
```
Comment (katkodominik) [inline resources/views/admin/internet/mac_addresses/list.blade.php]: The ids were swapped.

**Comment by katkodominik** [comment]
As I worked with this framework, I found out that maybe this was not the best choice (there are more powerful frameworks, and materialize didn't get any updates in a while). In the long run, it would be better to rewrite this to a better one. (Since the layout and the components are final, I hope it can be made easily).

**Comment by katkodominik** [comment]
> As I worked with this framework, I found out that maybe this was not the best choice (there are more powerful frameworks, and materialize didn't get any updates in a while). In the long run, it would be better to rewrite this to a better one. (Since the layout and the components are final, I hope it can be made easily).

I looked at several other options. In the end, for this design, I found materialize is the best (only typescript/js-based frameworks looks promising like Vue or Angular, but these are unnecessarily difficult to learn). I even looked back at bootstrap, but for example, bootstrap still hasn't got good sidebar, or built-in toast notifications which I want to use more in the future (and I still don't like their design, but it's my bad 🙃). Among the others, MDB is very promising, but it has premium subscriptions for features we need (e.g. searchable select). 

After all, I found some bugs which I was worried about and fixed them. The current features should work properly.
I found out that materialize have more talkative class names than bootstrap, and I think it's easier to work with. For now, I think we could be good without any framework updates, but I know it's not so comforting (the maintaners are just "taking a break", but the issues and pr's are coming continuously, so it is not that bad). What do you think?

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (luksan47) [reply app/Http/Controllers/Auth/RegisterController.php:73]: Is this the result of your change?

**Comment by luksan47** [comment]
> don't forget to clear your browser cache and also warn the users about this after releasing...

We could force browsers to clear cache by doing something like this: https://stackoverflow.com/questions/1922910/force-browser-to-clear-cache

I think we could write a script and add it to the release process.

**Comment by luksan47** [comment]
> new logo (may be changed, the other pictures are still in the directory, and we can even add more, because i don't like that this logo does not have the 'ec' colors :D )

We will discuss it in the evening. We should also decide if we release a new major version and rename the app.

File: app/Http/Controllers/InternetController.php [hunk 4]
```diff
@@ -82,10 +83,6 @@ public function deleteMacAddress(Request $request, $id)
 
     public function resetWifiPassword(Request $request)
     {
-        $request->validate([
-            'confirm' => 'accepted',
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply app/Http/Controllers/InternetController.php:86]: I deleted the "Nem baj" check if the user has acknowledged that he/she might need to relogin. I don't think it's elegant. If we need some confirmation, I can do it in other ways, but the warning message is still displayed.

File: app/Http/Controllers/PrintController.php [hunk 5]
```diff
@@ -89,13 +94,13 @@ public function print(Request $request) {
     public function transferBalance(Request $request) {
         $validator = Validator::make($request->all(), [
             'balance' => 'required|integer|min:1',
-            'receiver_id' => 'required|integer|exists:users,id'
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply app/Http/Controllers/PrintController.php:92]: Yes, in the template blade, the form refers to it otherwise. I can get some workarounds if needed.

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (luksan47) [reply app/Http/Controllers/Auth/RegisterController.php:73]: I think it was public/css/bootstrap-formhelpers.min.css and .js.
Could you try re-adding it and check if that solves the problem?

File: app/Http/Controllers/InternetController.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply app/Http/Controllers/InternetController.php:86]: Okay, we can leave that for now and see later if it's needed.

File: app/Http/Controllers/PrintController.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply app/Http/Controllers/PrintController.php:92]: Leave it as user_id for this PR, it's a big change already. We can change back when solving #152

File: resources/views/layouts/app.blade.php [hunk 6]
```diff
@@ -1,65 +1,78 @@
...
     <script src="{{ asset('js/popper.min.js') }}"></script>
-    <script src="{{ asset('js/bootstrap.min.js') }}"></script>
-    <script src="{{ asset('js/bootstrap-select.min.js') }}"></script>
-    <script src="{{ asset('js/bootstrap-datepicker.min.js') }}"></script>
-    <script src="{{ asset('js/bootstrap-formhelpers.min.js') }}"></script>
     <script src="{{ asset('js/tabulator.min.js') }}" defer></script>
     <script src="{{ asset('js/site.js') }}" defer></script>
     <script src="{{ asset('js/cookieconsent.min.js') }}" defer></script>
     <script src="{{ asset('js/cookieconsent-initialize.js') }}" defer></script>
+    <!-- modified materialize js for searchable select: https://codepen.io/yassinevic/pen/eXjqjb?editors=1111 -->
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/layouts/app.blade.php:35]: It is not my fix, I just found it. I can try to separate the changes from the original one.

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (kdmnk) [reply app/Http/Controllers/Auth/RegisterController.php:73]: The original one was only allowed +dd (dd) ddd-dddd which wasn't even suitable for international phone numbers. Now it will not format the text automatically, but the user needs to follow the formatting.

File: resources/views/welcome.blade.php [hunk 7]
```diff
@@ -1,108 +1,131 @@
...
+    <div class="flex-center position-ref full-height">
+        <div class="content">
+            <div class="row">
+                <div class="col s12 l2 offset-l2 center-align">
+                    <img class="material-icons" style="height:130px" src="{{ env('LOGO','') }}">
                 </div>
+                <div class="col s12 l5 center-align">
+                    <div class="noselect"
+                        style="text-indent:15px;font-size:80px;text-transform: uppercase;font-weight:300;letter-spacing:3px;">
+                        {{ config('app.name', 'Urán') }} </div>
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/welcome.blade.php:109]: But it is in the config. Or what do you mean?

File: resources/views/welcome.blade.php [hunk 8]
```diff
@@ -1,108 +1,131 @@
...
+            <nav class="top-nav">
+                <div class="nav-wrapper">
+                    <div class="row">
+                        <ul class="right">
+                            @auth
+                            <li><a href="{{ url('/home') }}">@lang('general.login')</a></li>
+                            @else
+                            <li><a href="{{ route('login') }}">@lang('general.login')</a></li>
+
+                            @if (Route::has('register'))
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/welcome.blade.php:85]: Idk. Those were there before. I deleted it, and it looks ok though...

File: resources/views/print/send/send.blade.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<div class="card">
+    <div class="card-content">
+        <span class="card-title">@lang('print.transfer_money')</span>
+        @if ($errors->any())
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/print/send/send.blade.php:4]: I will create a new issue for this since it`s a problem elsewhere also.

File: resources/views/print/print/print.blade.php [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        <blockquote>
+            @lang('print.available_money'): {{ Auth::user()->printAccount->balance }} HUF. 
+            @lang('print.upload_money')
+        </blockquote>
+        <form class="form-horizontal" role="form" method="POST" action="{{ route('print.print') }}"
+            enctype="multipart/form-data">
+            @csrf
+            @method('PUT')
+            <div class="row">
+                <div class="file-field input-field col s12 m12 l8 xl10">
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/print/print/print.blade.php:18]: What do you mean?

File: resources/views/internet/mac_addresses/list.blade.php [hunk 11]
```diff
@@ -1,37 +1,36 @@
...
+                    },
+                    error: function(error) {
+                        ajaxError('@lang('internet.error')', '@lang('internet.ajax_error')', '@lang('internet.ok')', error);
+                    }
                 });
             })[0];
         };
-
         var table = new Tabulator("#mac-addresses-table", {
-            paginationSizeSelector: [10, 25, 50, 100, 250, 500],
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/internet/mac_addresses/list.blade.php:23]: The paginationSizeSelector is not working properly with material style tabulators.. :/

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (luksan47) [reply app/Http/Controllers/Auth/RegisterController.php:73]: Yeah, but the users won't follow the formatting. :/
But since it allows more, I'm okay with it. Maybe setting the country code on a different field would be the best final solution; in another PR.

File: resources/views/welcome.blade.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/welcome.blade.php:85]: I think it was in pre-built Laravel, authentication is not there by default.

File: resources/views/welcome.blade.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/welcome.blade.php:109]: You can write this  in the config file (it's already there):
`'name' => env('APP_NAME', 'Urán'),`
This way you can use simply `config('app.name')`.

File: resources/views/print/send/send.blade.php (hunk 9 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/print/send/send.blade.php:4]: OK.

File: resources/views/print/print/print.blade.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/print/print/print.blade.php:18]: I mean s12 m12 l8 xl10.
Were there any reason that you used these?

File: resources/views/layouts/app.blade.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/layouts/app.blade.php:35]: It would be best if we could also document it somewhere.

File: resources/views/internet/mac_addresses/list.blade.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/internet/mac_addresses/list.blade.php:23]: OK.

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (kdmnk) [reply app/Http/Controllers/Auth/RegisterController.php:73]: > Yeah, but the users won't follow the formatting. :/
I mean the input will turn red if the formatting is not valid, so it only accepts valid phone numbers.

File: resources/views/print/print/print.blade.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/print/print/print.blade.php:18]: s stands for small, m: tablet, l: desktop, xl: large desktop
and altogether, there are 12 coloumns
So yes :)

[the remark this answers is not in the document]
Comment (luksan47) [reply resources/views/print/print/print.blade.php:18]: I know this, but my question is regarding if you have tested that these values look good on all devices?

File: app/Http/Controllers/Auth/RegisterController.php (hunk 1 above)
Comment (luksan47) [reply app/Http/Controllers/Auth/RegisterController.php:73]: Okay, I see now.

File: resources/views/print/print/print.blade.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (kdmnk) [reply resources/views/print/print/print.blade.php:18]: Yes. ;)

File: app/Http/Controllers/InternetController.php [hunk 12]
```diff
@@ -28,7 +28,8 @@ public function index()
     public function admin()
     {
         $activationDate = env('INTERNET_ACTIVATION_DATE'); //TODO: get date for current semester
-        return view('admin.internet.app', ['activation_date' => $activationDate]);
+
+        return view('admin.internet.app', ['activation_date' => $activationDate, 'users' => User::all()]);
```
[the remark this answers is not in the document]
Comment (kdmnk) [reply app/Http/Controllers/InternetController.php:32]: I might open a new issue for this. It is because the search-user function, and yes - it's unnecessary. 
Maybe the best solution would be if the search-user template requested the needed info with ajax.

[the remark this answers is not in the document]
Comment (luksan47) [reply app/Http/Controllers/InternetController.php:32]: Okay, let's open an issue for that.

File: app/Http/Controllers/Auth/RegisterController.php [hunk 13]
```diff
@@ -70,7 +70,7 @@ protected function validator(array $data)
             'place_of_birth' => ['required', 'string', 'max:255'],
             'date_of_birth' => ['required', 'date_format:Y-m-d'],
             'mothers_name' => ['required', 'string', 'max:255'],
-            'phone_number' => ['required', 'string', 'min:11', 'max:18'],
+            'phone_number' => ['required', 'string', 'min:16', 'max:18'],
```
Comment (kdmnk) [inline app/Http/Controllers/Auth/RegisterController.php]: changed it back to original

File: public/js/cookieconsent-initialize.js [hunk 14]
```diff
@@ -1,11 +1,11 @@
...
-            "text": "#404040"
+            "background": "#252A51",
+            "text": "#ffffff"
         },
         "button": {
-            "background": "#8ec760",
+            "background": "#B38F2F",
             "text": "#ffffff"
         }
     }
```
Comment (kdmnk) [inline public/js/cookieconsent-initialize.js:11]: Changed cookie message design also.

Comment (luksan47) [reply public/js/cookieconsent-initialize.js:11]: Nice, thanks!

File: config/app.php [hunk 15]
```diff
@@ -15,6 +15,14 @@
 
     'name' => env('APP_NAME', 'Urán'),
 
+    'version' => '3.0',
+
+    'logo_blue_path' => '/img/uran_blue.png',
```
Comment (luksan47) [inline config/app.php:20]: Do we need all of these here? This is a bit overcomplicated. You cold just create the config option 'logo' and set it's value to '/img/uran_blue.png'. If someone wants to change it, they will rewrite the value in the config.

Comment (kdmnk) [reply config/app.php:20]: But what about the other 2? I use them too.

Comment (luksan47) [reply config/app.php:20]: Okay, then it's fine.
